File: contact9/src/main/java/cn/ucai/contact9/controller/activity/MainActivity.java
package cn.ucai.contact9.controller.activity;

import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.ViewPropertyAnimator;
import android.widget.TextView;

import cn.ucai.contact9.R;
import cn.ucai.contact9.controller.fragments.ContactFragment;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    TextView mNewgoods,mBoutique,mCategory,mCart,mContact;

    ContactFragment fragment;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        setListener();
    }

    private void setListener() {
        mNewgoods.setOnClickListener(this);
        mBoutique.setOnClickListener(this);
        mCategory.setOnClickListener(this);
        mCart.setOnClickListener(this);
        mContact.setOnClickListener(this);
    }

    private void initView() {
        mNewgoods = (TextView) findViewById(R.id.tvNewgoods);
        mBoutique = (TextView) findViewById(R.id.tvBoutique);
        mCategory = (TextView) findViewById(R.id.tvCategory);
        mCart = (TextView) findViewById(R.id.tvCart);
        mContact = (TextView) findViewById(R.id.tvContact);

        fragment = new ContactFragment();

    }
    public void startRotate(View v){
        float y = 360-v.getRotationY();
        ViewPropertyAnimator animate = v.animate();
        animate.setDuration(1000).rotationY(y).start();
    }

    @Override
    public void onClick(View v) {
        FragmentManager manager = getSupportFragmentManager();
        FragmentTransaction ft = manager.beginTransaction();
        initDrawable();
        switch (v.getId()){
            case R.id.tvNewgoods:
                setDrawable(mNewgoods,Color.rgb(0xff,0x66,0xff),R.drawable.menu_item_new_goods_selected);
                startRotate(v);
                break;
            case R.id.tvBoutique:
                setDrawable(mBoutique,Color.rgb(0xff,0x66,0xff),R.drawable.menu_item_boutique_selected);
                startRotate(v);
                break;
            case R.id.tvCategory:
                setDrawable(mCategory,Color.rgb(0xff,0x66,0xff),R.drawable.menu_item_category_selected);
                startRotate(v);
                break;
            case R.id.tvCart:
                setDrawable(mCart,Color.rgb(0xff,0x66,0xff),R.drawable.menu_item_cart_selected);
                startRotate(v);
                break;
            case R.id.tvContact:
                setDrawable(mContact,Color.rgb(0xff,0x66,0xff),R.drawable.menu_item_contact_selected);
                startRotate(v);
                ft.replace(R.id.frameContact, fragment).commit();
                break;
        }

    }
    public void initDrawable(){
        setDrawable(mNewgoods, Color.GRAY,R.drawable.menu_item_new_goods_normal);
        setDrawable(mBoutique, Color.GRAY,R.drawable.menu_item_boutique_normal);
        setDrawable(mCategory, Color.GRAY,R.drawable.menu_item_category_normal);
        setDrawable(mCart, Color.GRAY,R.drawable.menu_item_cart_normal);
        setDrawable(mContact, Color.GRAY,R.drawable.menu_item_contact_normal);

    }
    public void setDrawable(TextView tv,int color,int drawable){
        tv.setTextColor(color);
        Drawable top = ContextCompat.getDrawable(this, drawable);
        top.setBounds(0, 0, top.getMinimumWidth(), top.getMinimumHeight());
        tv.setCompoundDrawables(null,top,null,null);
    }
}

File: contact8/src/main/java/cn/ucai/contact8/MainActivity.kt
package cn.ucai.contact8

import android.graphics.Color
import android.graphics.drawable.Drawable
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentTransaction
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.ViewPropertyAnimator
import android.widget.TextView

import cn.ucai.contact8.fragments.ContactFragment
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.onClick

class MainActivity : AppCompatActivity(), View.OnClickListener {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setListener()
    }

    private fun setListener() {
        tvNewGoods.onClick {
            startRotate(tvNewGoods)
            setDrawable(tvNewGoods, Color.rgb(0xff, 0x66, 0xff), R.drawable.menu_item_new_goods_selected)
        }
        tvBoutique.onClick { this }
        tvCategory.onClick { this }
        tvCart.onClick { this }
        tvContact.onClick { this }
    }

    fun startRotate(v: View) {
        val y = 360 - v.rotationY
        val animate = v.animate()
        animate.setDuration(1000).rotationY(y).start()

    }

    override fun onClick(v: View) {
        val manager = supportFragmentManager
        val ft = manager.beginTransaction()
        initDrawable()
        when (v.id) {
            R.id.tvBoutique -> {
                startRotate(v)
                setDrawable(tvBoutique, Color.rgb(0xff, 0x66, 0xff), R.drawable.menu_item_boutique_selected)
            }
            R.id.tvCategory -> {
                startRotate(v)
                setDrawable(tvCategory, Color.rgb(0xff, 0x66, 0xff), R.drawable.menu_item_category_selected)
            }
            R.id.tvCart -> {
                startRotate(v)
                setDrawable(tvCart, Color.rgb(0xff, 0x66, 0xff), R.drawable.menu_item_cart_selected)
            }
            R.id.tvContact -> {
                startRotate(v)
                setDrawable(tvContact, Color.rgb(0xff, 0x66, 0xff), R.drawable.menu_item_contact_selected)
                ft.replace(R.id.frame_contact, ContactFragment()).commit()
            }
        }

    }

    fun initDrawable() {
        setDrawable(tvNewGoods, Color.GRAY, R.drawable.menu_item_new_goods_normal)
        setDrawable(tvBoutique, Color.GRAY, R.drawable.menu_item_boutique_normal)
        setDrawable(tvCategory, Color.GRAY, R.drawable.menu_item_category_normal)
        setDrawable(tvCart, Color.GRAY, R.drawable.menu_item_cart_normal)
        setDrawable(tvContact, Color.GRAY, R.drawable.menu_item_contact_normal)

    }

    fun setDrawable(tv: TextView, color: Int, drawable: Int) {
        tv.setTextColor(color)
        val top = ContextCompat.getDrawable(this, drawable)
        top.setBounds(0, 0, top.minimumWidth, top.minimumHeight)
        tv.setCompoundDrawables(null, top, null, null)
    }
}
